lua/presence/init.lua

```diff
@@ -466,6 +466,12 @@ function Presence:update_for_buffer(buffer, should_debounce)
     }
 
     local editing_text = self.options.editing_text
+    if vim.bo.modifiable then
+        editing_text = "Editing %s"
+    else
+        editing_text = "Reading %s"
+    end
```

Comment:
Currently, the `editing_text` option can be either a static format string or a function with the signature `function(filename, buffer): string`.

Since we're moving away from a static `"Editing %s"` string to a dynamic one, it'd be nice to have this logic in some `get_editing_text` function to be set on the option instead.

What do you think about moving these lines to a new method and updating the existing [`set_option`](https://github.com/kylechui/presence.nvim/blob/main/lua/presence/init.lua#L91) call to use that method as the default value argument? Pseudocode would probably be something like:
```lua
function Presence.get_editing_text(filename)
    if modifiable and not readonly then
        return string.format("Editing %s", filename)
    else
        return string.format("Reading %s", filename)
    end
end

function Presence:setup(options)
    -- ... existing setup code...
    self:set_option("editing_text", self.get_editing_text)
    -- ... more setup code...
end
```

Reply:
Sorry, I saw this---I think a new method sounds like the best way of going about things. I'll try implementing this right now

Reply:
Hmm actually that's a good idea, but considering that people may already be using the `editing_text` option I wouldn't want to release a breaking change without first having some intermediary deprecation warning.

That may be a bit out of scope for this feature, so we can keep it as `editing_text` for now and I'll implement deprecation warnings after this feature's merged in!